pkgs/applications/networking/remote/xrdp/pulseaudio-module-xrdp/default.nix

```diff
@@ -0,0 +1,76 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, nix-update-script
+, pulseaudio
+, autoreconfHook
+, pkg-config
+}:
+
+let
+
+  # this plugin requires some files from pulseaudio that are only available after configure phase
+  # the purpose of this derivation is to get the nearest possible environment used in pulseaudio
+  # and export this data
```

Comment:
Wouldn't it make more sense to edit the pulseaudio derivation to provide these files? (Maybe in a new output?)

Reply:
No idea. BTW it's a very specific dependency. IMO it doesn't make sense to opt to make pulseaudio bigger because of a very specific usecase. BTW, this module doesn't require pulseaudio to be fully compiled, that source workaround is only to get some files that the module needs, like config.h.

Reply:
We could have a separate output for PulseAudio here, e.g.: `dev` for those cases.

Looks like a better alternative than the current one.

Reply:
We actually already have a `dev` output in `pulseaudio`. Does it work here?

Reply:
If it doesn't, maybe creating a `src` output?

Reply:
If it doesn't maybe using `pulseaudio.src` works here?

Reply:
> We could have a separate output for PulseAudio here, e.g.: `dev` for those cases.
> 
> Looks like a better alternative than the current one.

I tried that and it doesn't work. It needs both the sources and config.h so I did that override to generate the most precise config.h without having to compile pulseaudio fully. Otherwise it doesn't work or may have issues when the pulseaudio derivation is modified.

Reply:
Maybe just copy the `pulseaudio.src` + whatever files you need from `pulseaudio` derivation itself using `runCommand` them? Because overriding the derivation the way you did is really fragile, and also will incur an unnecessary configuration step.

Reply:
The configuration step is required, another build of pulseaudio don't.

config.h is generated in this phase and required for the plugin.

Reply:
I'd still prefer a better solution than creating a pulseaudio variant hidden here away from where the pulseaudio maintainers know about it.

Reply:
I exposed config.h as an extra pulseaudio output.

Reply:
And by consequence I caused a mass rebuild lol